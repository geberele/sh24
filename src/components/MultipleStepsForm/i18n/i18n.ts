import { I18N } from './i18n.types'

export const en: I18N = {
  question1: 'What is your name?',
  question2: 'What is your email address?',
  question3: 'What service are you here for?',
  options1: 'STI Testing',
  options2: 'Contraception',
  options3: 'Other',
  labelName: 'Name',
  labelEmail: 'Email address',
  previewTitle: 'Preview',
  prev: 'Back',
  next: 'Next',
  errorMessage1: 'Please insert a valid name (2-25 characters)',
  errorMessage2: 'Please insert a valid email',
  errorMessage3: 'Please select an option',
}

export const it: I18N = {
  question1: 'Come ti chiami?',
  question2: 'Quale Ã¨ la tua email?',
  question3: 'Che tipo di servizio ti interessa?',
  options1: 'Test STI',
  options2: 'Contraccettivo',
  options3: 'Altro',
  labelName: 'Nome',
  labelEmail: 'Email',
  previewTitle: 'Preview',
  prev: 'Indietro',
  next: 'Avanti',
  errorMessage1: 'Inserisci un nome valido (2-25 lettere)',
  errorMessage2: 'Inserisci un\'email valida',
  errorMessage3: 'Seleziona un\'opzione per favore',
}
