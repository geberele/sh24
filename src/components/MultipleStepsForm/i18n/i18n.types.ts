export interface I18N {
  question1: string
  question2: string
  question3: string
  options1: string
  options2: string
  options3: string
  labelName: string
  labelEmail: string
  previewTitle: string
  prev: string
  next: string
  errorMessage1: string
  errorMessage2: string
  errorMessage3: string
}
