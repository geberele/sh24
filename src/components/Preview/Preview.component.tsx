import React, { Fragment } from 'react'

export interface PreviewProps {
  title: string
  qandas: QandA[]
  prev: string
  prevStep: () => void
}

export interface QandA {
  question: string
  answer: string
}

export const Preview = ({
  title,
  qandas,
  prev,
  prevStep,
}: PreviewProps) => (
  <>
    <h1 data-testid="preview-title">{title}</h1>
    {qandas.map(({ question, answer }, key) => (
      <Fragment key={key}>
        <h2>{question}</h2>
        <p>{answer}</p>
      </Fragment>
    ))}

    <div className="navigation">
      <button
        onClick={prevStep}
        className="navigationLeft"
        data-testid="button-prev"
      >
        {prev}
      </button>
    </div>
  </>
)
