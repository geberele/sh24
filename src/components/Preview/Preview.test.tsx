import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { fireEvent, waitFor, screen } from '@testing-library/dom'
import { Preview, PreviewProps, QandA } from './Preview.component'
import { en } from '../MultipleStepsForm/i18n/i18n'

describe('Preview', () => {
  const prevStep = jest.fn()

  const name = 'Mark'
  const email = 'test@example.com'
  const service = 'Other'
  const qandas: QandA[] = [
    { question: en.question1, answer: name },
    { question: en.question2, answer: email },
    { question: en.question3, answer: service },
  ]

  const props: PreviewProps = {
    title: en.previewTitle,
    prev: en.prev,
    qandas: qandas,
    prevStep: prevStep,
  }

  const setup = () => {
    const utils = render(<Preview {...props}/>)
    const prevButton = screen.getByTestId('button-prev') as HTMLInputElement
    return {
      prevButton,
      ...utils,
    }
  }

  it('renders', () => {
    const { asFragment } = setup()
    expect(asFragment()).toMatchSnapshot()
  })

  it('shows the correct answers', async () => {
    setup()
    expect(screen.getByText(name)).toBeInTheDocument()
    expect(screen.getByText(email)).toBeInTheDocument()
    expect(screen.getByText(service)).toBeInTheDocument()
  })

  it('goes to the previous step', async () => {
    const { prevButton } = setup()
    fireEvent.click(prevButton)
    await waitFor(() => expect(prevStep).toHaveBeenCalledTimes(1))
  })
})
