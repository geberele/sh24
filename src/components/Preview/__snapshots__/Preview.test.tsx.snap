// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview renders 1`] = `
<DocumentFragment>
  <h1>
    Preview
  </h1>
  <h2>
    What is your name?
  </h2>
  <p>
    Mark
  </p>
  <h2>
    What is your email address?
  </h2>
  <p>
    test@example.com
  </p>
  <h2>
    What service are you here for?
  </h2>
  <p>
    Other
  </p>
  <div
    class="navigation"
  >
    <button
      class="navigationLeft"
      data-testid="button-prev"
    >
      Back
    </button>
  </div>
</DocumentFragment>
`;
